### PR TITLE
docs(sdk): fix docstrings to enable project-wide pydocstyle checks

### PR DESCRIPTION
**Pull request by moredatarequired**
Description
-----------
What is this abomination?

This fixes a bunch of documentation lint errors all over our repo; after this PR, pycodestyle runs without error as long as we skip `D1xx` (don't require docstrings, only check ones that exist) and `D417` (don't require documenting all arguments). This means we no longer have to keep a short allowlist of files to run docstyle checks on and can start keeping documentation up to standards as we go.

This PR was a lot of auto-fix (`ruff` can fix a lot on its own!), but way more manual editing than I care to admit. Some `sed` work as well, so there might be a few mistakes I made and didn't catch. But I did manually review every. single. change. and fixed all the mistakes I did catch. In any case, this is not a functional change because it only touches docstrings.

For posterity, this PR fixes
| Count | Code | Description |
| ----- | ---- | ----------- |
| 123   | D200 | One-line docstring should fit on one line with quotes |
| 45    | D202 | No blank lines allowed after function docstring |
| 224   | D205 | blank line required between summary line and description |
| 4     | D208 | Docstring is over-indented |
| 24    | D209 | Multi-line docstring closing quotes should be on a separate line |
| 300   | D212 | Summary should start on first line |
| 2     | D300 | Use """triple double quotes""" |
| 4     | D301 | Use r""" if any backslashes in a docstring |
| 1     | D402 | First line should not be the function's signature |
| 16    | D403 | First word of the first line should be properly capitalized |
| 3     | D405 | Section name should be properly capitalized |
| 6     | D410 | Missing blank line after section |
| 17    | D411 | Missing blank line before section |
| 419   | D415 | First line should end with a period |
| 5     | D416 | Section name should end with a newline |

Testing
-------
No functional changes, just relying on CI.

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: tests/functional_tests/t0_main/core/t8_init_fail.py [hunk 1]
```diff
@@ -15,7 +16,7 @@
   - WANDB_API_KEY: thisisnotavalidkey
 assert:
   - :wandb:runs_len: 0
-  - :yea:exit: 0
+  - :yea:exit: 0.
```
Comment (moredatarequired) [inline tests/functional_tests/t0_main/core/t8_init_fail.py:19]:
```suggestion
  - :yea:exit: 0
```

File: tests/functional_tests/t0_main/sklearn/feature_importance_attribute_does_not_exist.py [hunk 2]
```diff
@@ -15,7 +16,7 @@
     - :yea:exit: 0
     - :op:contains_regex:
       - :wandb:runs[0][output][stderr]
-      - Cannot plot feature importances
+      - Cannot plot feature importances.
```
Comment (moredatarequired) [inline tests/functional_tests/t0_main/sklearn/feature_importance_attribute_does_not_exist.py:19]:
```suggestion
      - Cannot plot feature importances
```

File: tests/functional_tests/t0_main/sklearn/feature_importance_attribute_exists_for_elasticnet.py [hunk 3]
```diff
@@ -15,7 +16,7 @@
     - :yea:exit: 0
     - :wandb:runs[0][summary][feature_importances][_type]: table-file
     - :wandb:runs[0][summary][feature_importances][ncols]: 2
-    - :wandb:runs[0][summary][feature_importances][nrows]: 30
+    - :wandb:runs[0][summary][feature_importances][nrows]: 30.
```
Comment (moredatarequired) [inline tests/functional_tests/t0_main/sklearn/feature_importance_attribute_exists_for_elasticnet.py:19]:
```suggestion
    - :wandb:runs[0][summary][feature_importances][nrows]: 30
```

File: tests/functional_tests/t0_main/sklearn/feature_importance_attribute_exists_for_randomforest.py [hunk 4]
```diff
@@ -15,7 +16,7 @@
     - :yea:exit: 0
     - :wandb:runs[0][summary][feature_importances][_type]: table-file
     - :wandb:runs[0][summary][feature_importances][ncols]: 2
-    - :wandb:runs[0][summary][feature_importances][nrows]: 30
+    - :wandb:runs[0][summary][feature_importances][nrows]: 30.
```
Comment (moredatarequired) [inline tests/functional_tests/t0_main/sklearn/feature_importance_attribute_exists_for_randomforest.py:19]:
```suggestion
    - :wandb:runs[0][summary][feature_importances][nrows]: 30
```

File: tests/functional_tests/t0_main/sklearn/plot_classifier-basic.py [hunk 5]
```diff
@@ -31,7 +32,7 @@
     - :wandb:runs[0][summary][roc][ncols]: 3
     - :wandb:runs[0][summary][summary_metrics][_type]: table-file
     - :wandb:runs[0][summary][summary_metrics][ncols]: 3
-    - :wandb:runs[0][summary][summary_metrics][nrows]: 4
+    - :wandb:runs[0][summary][summary_metrics][nrows]: 4.
```
Comment (moredatarequired) [inline tests/functional_tests/t0_main/sklearn/plot_classifier-basic.py:35]:
```suggestion
    - :wandb:runs[0][summary][summary_metrics][nrows]: 4
```

File: tests/functional_tests/t0_main/sklearn/plot_clusterer-basic.py [hunk 6]
```diff
@@ -17,7 +18,7 @@
     - :wandb:runs[0][summary][elbow_curve][ncols]: 3
     - :wandb:runs[0][summary][silhouette_plot][_type]: table-file
     - :wandb:runs[0][summary][silhouette_plot][ncols]: 10
-    - :yea:exit: 0
+    - :yea:exit: 0.
```
Comment (moredatarequired) [inline tests/functional_tests/t0_main/sklearn/plot_clusterer-basic.py:21]:
```suggestion
    - :yea:exit: 0
```

File: tests/functional_tests/t0_main/sklearn/plot_learning_curve-basic.py [hunk 7]
```diff
@@ -15,7 +16,7 @@
     - :yea:exit: 0
     - :wandb:runs[0][summary][learning_curve][_type]: table-file
     - :wandb:runs[0][summary][learning_curve][ncols]: 3
-    - :wandb:runs[0][summary][learning_curve][nrows]: 10
+    - :wandb:runs[0][summary][learning_curve][nrows]: 10.
```
Comment (moredatarequired) [inline tests/functional_tests/t0_main/sklearn/plot_learning_curve-basic.py:19]:
```suggestion
    - :wandb:runs[0][summary][learning_curve][nrows]: 10
```

File: tests/functional_tests/t0_main/sklearn/plot_regressor-basic.py [hunk 8]
```diff
@@ -22,7 +23,7 @@
     - :wandb:runs[0][summary][residuals][ncols]: 5
     - :wandb:runs[0][summary][summary_metrics][_type]: table-file
     - :wandb:runs[0][summary][summary_metrics][ncols]: 3
-    - :wandb:runs[0][summary][summary_metrics][nrows]: 3
+    - :wandb:runs[0][summary][summary_metrics][nrows]: 3.
```
Comment (moredatarequired) [inline tests/functional_tests/t0_main/sklearn/plot_regressor-basic.py:26]:
```suggestion
    - :wandb:runs[0][summary][summary_metrics][nrows]: 3
```

File: tests/functional_tests/t0_main/sklearn/plot_summary_metrics-basic.py [hunk 9]
```diff
@@ -15,7 +16,7 @@
     - :yea:exit: 0
     - :wandb:runs[0][summary][summary_metrics][_type]: table-file
     - :wandb:runs[0][summary][summary_metrics][ncols]: 3
-    - :wandb:runs[0][summary][summary_metrics][nrows]: 4
+    - :wandb:runs[0][summary][summary_metrics][nrows]: 4.
```
Comment (moredatarequired) [inline tests/functional_tests/t0_main/sklearn/plot_summary_metrics-basic.py:19]:
```suggestion
    - :wandb:runs[0][summary][summary_metrics][nrows]: 4
```

File: tests/pytest_tests/system_tests/test_core/test_wandb_integration.py [hunk 10]
```diff
@@ -76,13 +78,13 @@ def test_ignore_globs_wandb_files(relay_server, wandb_init):
...
 def _remove_dir_if_exists(path):
-    """Recursively removes directory. Be careful"""
+    """Recursively removes directory. Be careful."""
     if os.path.isdir(path):
         shutil.rmtree(path)
 
 
 def test_dir_on_import():
-    """Ensures that `import wandb` does not create a local storage directory"""
+    """Ensure hat `import wandb` does not create a local storage directory."""
```
Comment (moredatarequired) [inline tests/pytest_tests/system_tests/test_core/test_wandb_integration.py:87]:
```suggestion
    """Ensure that `import wandb` does not create a local storage directory."""
```

File: tests/pytest_tests/unit_tests/test_data_types.py [hunk 11]
```diff
@@ -680,13 +683,13 @@ def test_molecule_from_smiles(mock_run):
...
 def test_molecule_from_invalid_smiles():
-    """Ensures that wandb.Molecule.from_smiles errs if passed an invalid SMILES string"""
+    """Ensure that wandb.Molecule.from_smiles errs if passed an invalid SMILES string."""
     with pytest.raises(ValueError):
         wandb.Molecule.from_smiles("TEST")
 
 
 def test_molecule_from_rdkit_mol_object(mock_run):
-    """Ensures that wandb.Molecule.from_rdkit supports rdkit.Chem.rdchem.Mol objects"""
+    """Ensures that wandb.Molecule.from_rdkit supports rdkit.Chem.rdchem.Mol objects."""
```
Comment (moredatarequired) [inline tests/pytest_tests/unit_tests/test_data_types.py:692]:
```suggestion
    """Ensure that wandb.Molecule.from_rdkit supports rdkit.Chem.rdchem.Mol objects."""
```

File: tests/pytest_tests/unit_tests/test_data_types.py [hunk 12]
```diff
@@ -696,7 +699,7 @@ def test_molecule_from_rdkit_mol_object(mock_run):
 
 
 def test_molecule_from_rdkit_mol_file(mock_run):
-    """Ensures that wandb.Molecule.from_rdkit supports .mol files"""
+    """Ensures that wandb.Molecule.from_rdkit supports .mol files."""
```
Comment (moredatarequired) [inline tests/pytest_tests/unit_tests/test_data_types.py:702]:
```suggestion
    """Ensure that wandb.Molecule.from_rdkit supports .mol files."""
```

File: tests/pytest_tests/unit_tests/test_data_types.py [hunk 13]
```diff
@@ -709,7 +712,7 @@ def test_molecule_from_rdkit_mol_file(mock_run):
 
 
 def test_molecule_from_rdkit_invalid_input():
-    """Ensures that wandb.Molecule.from_rdkit errs on invalid input"""
+    """Ensures that wandb.Molecule.from_rdkit errs on invalid input."""
```
Comment (moredatarequired) [inline tests/pytest_tests/unit_tests/test_data_types.py:715]:
```suggestion
    """Ensure that wandb.Molecule.from_rdkit errs on invalid input."""
```

File: tests/pytest_tests/unit_tests/test_library_public.py [hunk 14]
```diff
@@ -1,9 +1,8 @@
-"""
-Library public tests.
+"""Library public tests.
 
 *NOTE*: If you need to add a symbol, make sure this has been discussed and the name of the object or method is agreed upon.
 
-TODO:
+Todo:
```
Comment (moredatarequired) [inline tests/pytest_tests/unit_tests/test_library_public.py:5]:
```suggestion
TODO:
```

File: tests/standalone_tests/artifact_load.py [hunk 15]
```diff
@@ -17,7 +17,7 @@
       - ensure files are available in bucket and match md5 / etag
       - ensure there are no dangling files in bucket
 
-TODO:
+Todo:
```
Comment (moredatarequired) [inline tests/standalone_tests/artifact_load.py:20]:
```suggestion
TODO:
```

File: wandb/data_types.py [hunk 16]
```diff
@@ -1,4 +1,4 @@
-"""This module defines data types for logging rich, interactive visualizations to W&B.
+"""Thi module defines data types for logging rich, interactive visualizations to W&B.
```
Comment (moredatarequired) [inline wandb/data_types.py:1]:
```suggestion
"""This module defines data types for logging rich, interactive visualizations to W&B.
```

File: wandb/sdk/launch/_project_spec.py [hunk 17]
```diff
@@ -226,7 +225,7 @@ def get_single_entry_point(self) -> Optional["EntryPoint"]:
         return list(self._entry_points.values())[0]
 
     def add_entry_point(self, command: List[str]) -> "EntryPoint":
-        """Adds an entry point to the project."""
+        """Add n entry point to the project."""
```
Comment (moredatarequired) [inline wandb/sdk/launch/_project_spec.py:228]:
```suggestion
        """Add an entry point to the project."""
```

File: wandb/sdk/lib/ipython.py [hunk 18]
```diff
@@ -52,7 +52,7 @@ def display_html(html: str):  # type: ignore
 
 
 def display_widget(widget):
-    """Displays ipywidgets in notebooks, is a noop outside of a jupyter context"""
+    """Displays ipywidgets in notebooks, is a noop outside of a jupyter context."""
```
Comment (moredatarequired) [inline wandb/sdk/lib/ipython.py:55]:
```suggestion
    """Display ipywidgets in notebooks, is a noop outside of a jupyter context."""
```

File: wandb/sdk/lib/ipython.py [hunk 19]
```diff
@@ -99,7 +99,7 @@ def close(self) -> None:
 
 
 def jupyter_progress_bar(min: float = 0, max: float = 1.0) -> Optional[ProgressWidget]:
-    """Returns an ipywidget progress bar or None if we can't import it"""
+    """Returns an ipywidget progress bar or None if we can't import it."""
```
Comment (moredatarequired) [inline wandb/sdk/lib/ipython.py:102]:
```suggestion
    """Return an ipywidget progress bar or None if we can't import it."""
```

File: wandb/sdk/wandb_run.py [hunk 20]
```diff
@@ -1830,7 +1830,7 @@ def restore(
     def finish(
         self, exit_code: Optional[int] = None, quiet: Optional[bool] = None
     ) -> None:
-        """Marks a run as finished, and finishes uploading all data.
+        """Mark a run as finished, and finishes uploading all data.
```
Comment (moredatarequired) [inline wandb/sdk/wandb_run.py:1833]:
```suggestion
        """Mark a run as finished, and finishe uploading all data.
```

File: wandb/sdk/wandb_run.py [hunk 21]
```diff
@@ -996,7 +996,7 @@ def _get_group(self) -> str:
     @property
     @_run_decorator._attach
     def group(self) -> str:
-        """Returns the name of the group associated with the run.
+        """Return the name of the group associated with the run.
```
Comment (moredatarequired) [inline wandb/sdk/wandb_run.py:999]:
```suggestion
        """Name of the group associated with the run.
```

File: wandb/sdk/wandb_run.py [hunk 22]
```diff
@@ -1016,7 +1016,7 @@ def job_type(self) -> str:
     @property
     @_run_decorator._attach
     def project(self) -> str:
-        """Returns the name of the W&B project associated with the run."""
+        """Return the name of the W&B project associated with the run."""
```
Comment (moredatarequired) [inline wandb/sdk/wandb_run.py:1019]:
```suggestion
        """Name of the W&B project associated with the run."""
```

**Review by moredatarequired — COMMENTED** [review]
Some additional changes (mistakes from initial sed passes?)

**Review by kptkin — APPROVED** [review]
Much appreciated changes!

**Review by jamie-rasmussen — APPROVED** [review]

**Review by ngrayluna — APPROVED** [review]
Simply amazing.

**Pull request merged**